### PR TITLE
refactor: use React.useContext replace ConfigConsumer

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor: use React.useContext replace ConfigConsumer |
| 🇨🇳 Chinese | 无需纳入 Change Log |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/AnchorLink.tsx [hunk 1]
```diff
@@ -46,33 +46,31 @@ const AnchorLink: React.FC<AnchorLinkProps> = (props) => {
     );
   }
 
+  const { getPrefixCls } = React.useContext<ConfigConsumerProps>(ConfigContext);
```
Comment (MadCcc) [inline components/anchor/AnchorLink.tsx:49]:
```suggestion
  const { getPrefixCls } = React.useContext(ConfigContext);
```
这里让类型自动推断吧

File: components/anchor/AnchorLink.tsx [hunk 2]
```diff
@@ -46,33 +46,31 @@ const AnchorLink: React.FC<AnchorLinkProps> = (props) => {
...
+
+  const titleClassName = classNames(`${prefixCls}-link-title`, {
+    [`${prefixCls}-link-title-active`]: activeLink === href,
+  });
+
   return (
-    <ConfigConsumer>
-      {({ getPrefixCls }: ConfigConsumerProps) => {
-        const prefixCls = getPrefixCls('anchor', customizePrefixCls);
-        const active = activeLink === href;
```
Comment (MadCcc) [inline components/anchor/AnchorLink.tsx:53]: 这行留一留

File: components/anchor/AnchorLink.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/anchor/AnchorLink.tsx]: done~

File: components/anchor/AnchorLink.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/anchor/AnchorLink.tsx:53]: done~

**Review by afc163 — APPROVED** [review]

**Pull request merged**